### ISSUE TITLE
Reorganize Modules Folder Structure

### ISSUE THREAD

**Issue by ccpandhare** [publiclab/image-sequencer#59]
I was implementing FishEyeGL and noticed that we need a different folder structure for that.
FishEyeGL has shaders. So there have to be multiple files.
So instead of this:
```
.
└─ src
    └─ modules
        ├─ Crop.js
        ├─ NdviRed.js
        └─ FishEyeGL.js
```
Should we move into this:
```
.
└─ src
    └─ modules
        ├─ Crop
        │   └─ Module.js
        ├─ NdviRed
        │   └─ Module.js
        └─ FishEyeGL
            ├─ Module.js (Connects Image Sequencer to the external module - fisheyegl.js)
            ├─ fisheyegl.js
            └─ shaders
                ├─ fragment.glfs
                ├─ method.glfs
                └─ vertex.glfs
```

Benfits of this folder structure :
* Multiple files per module can be stored.
* Easier to port external modules - We can have the external module file and our own `Module.js` which will act as an API between the external module file and hence convert the external module into an ImageSequencer "Module"
* Easier to maintain long code for complex modules like "segmented-colormap" (I know, this isn't that complex, but modularization will make the code more readable.)

**Comment by ccpandhare** [comment]
We can then also have a separate documentation file for each module in their own folder. Not sure if this is useful...

**Comment by jywarren** [comment]
I like this. Esp for the different docs files in each folder. Eventually
this also makes it easier to break modules out into their own NPM packages
if we want to do that.

**Comment by ccpandhare** [comment]
Okay then I shall work on this!

**Comment by ccpandhare** [comment]
Fixed in #60
